### PR TITLE
Fix service worker

### PR DESCRIPTION
**Pull request by kfarnung**
CRA v4 moved the service worker out of the core and into a template.
Incorporate the template into the app.

https://github.com/facebook/create-react-app/issues/9776

### CODE REVIEW

**Comment by kfarnung** [comment]
Deployed this to https://kfarnung.github.io/puzztool/ and it seems to be working now.

File: src/service-worker.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const fileExtensionRegexp = new RegExp("/[^/?]+\\.[^/]+$");
+registerRoute(
+  // Return false to exempt requests from being fulfilled by index.html.
+  ({ request, url }: { request: Request; url: URL }) => {
+    // If this isn't a navigation, skip.
+    if (request.mode !== "navigate") {
+      return false;
+    }
+
+    // If this is a URL that starts with /_, skip.
```
Comment (beckbria) [inline src/service-worker.ts:39]: This comment doesn't tell me anything the code doesn't.  Consider explaining what /_ signifies

File: src/reportWebVitals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from "web-vitals";
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import("web-vitals").then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+      getCLS(onPerfEntry);
```
Comment (beckbria) [inline src/reportWebVitals.ts:6]: Nit: Consider adding comments explaining this acronyms

File: src/service-worker.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      return false;
+    }
+
+    // Return true to signal that we want to use the handler.
+    return true;
+  },
+  createHandlerBoundToURL(process.env.PUBLIC_URL + "/index.html")
+);
+
+// An example runtime caching route for requests that aren't handled by the
```
Comment (beckbria) [inline src/service-worker.ts:56]: Do we actually use anything in this block?

**Review by beckbria — APPROVED** [review]

File: src/service-worker.ts (hunk 1 above)
Comment (kfarnung) [reply src/service-worker.ts:39]: This is copied from the PWA template, I'll leave it as-is for now.

File: src/reportWebVitals.ts (hunk 2 above)
Comment (kfarnung) [reply src/reportWebVitals.ts:6]: This is copied from the PWA template, I'll leave it as-is for now.

File: src/service-worker.ts (hunk 3 above)
Comment (kfarnung) [reply src/service-worker.ts:56]: I'm honestly not sure on this, I just grabbed this from the PWA template.

**Pull request merged**